crates/aptos-runtimes/src/thread_manager.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{spawn_named_runtime_with_start_hook, spawn_rayon_thread_pool_with_start_hook};
+#[cfg(target_os = "linux")]
+use libc::{cpu_set_t, sched_setaffinity, CPU_SET};
+use once_cell::sync::{Lazy, OnceCell};
+use rayon::ThreadPool;
+use tokio::runtime::{Handle, Runtime};
+
+pub static THREAD_MANAGER: Lazy<ThreadManager> =
+    Lazy::new(|| ThreadManager::new(ThreadManager::get_thread_config_strategy()));
+
+static THREAD_CONFIG_STRATEGY: OnceCell<ThreadConfigStrategy> = OnceCell::new();
+
+struct ThreadsConfig {
+    num_threads: usize,
+    on_thread_start: Box<dyn Fn() + Send + Sync + 'static>,
+}
+
+impl Default for ThreadsConfig {
+    fn default() -> Self {
+        Self::new_without_start_hook(num_cpus::get())
+    }
+}
+
+impl ThreadsConfig {
+    fn new_without_start_hook(num_threads: usize) -> Self {
+        Self::new(num_threads, || {})
+    }
+
+    fn new<F>(num_threads: usize, on_thread_start: F) -> Self
+    where
+        F: Fn() + Send + Sync + 'static,
+    {
+        Self {
+            num_threads,
+            on_thread_start: Box::new(on_thread_start),
+        }
+    }
+}
+
+#[derive(Default)]
+struct ThreadsConfigs {
+    tokio_config: ThreadsConfig,
+    rayon_config: ThreadsConfig,
+}
+
+struct Threads {
+    runtime: Runtime,
+    cpu_pool: ThreadPool,
+}
+
+pub struct ThreadManager {
+    exe_threads: Threads,
```

Comment:
Just a temporary way to separate pools for threads that potentially do IO v.s. threads that only do computation. 

I will need to do some experiments to find out a good strategy, it might end up in a totally different way. This PR is just to setup a playground so people can start playing with it.